src/LayerEditor/ui/panels/regions_panel.py

```diff
@@ -466,13 +461,18 @@ def _name_editing_finished(self):
         :return:
         """
         new_name = self.wg_name.text().strip()
-        if  new_name in self.layer_heads.region_names and new_name != self.curr_region.name:
+        if new_name == self.curr_region.name:
+            return
+        elif  new_name in self.layer_heads.region_names:
             error = "Region name already exist"
         elif not new_name:
             error = "Region name is empty"
         else:
             error = None
-            self.curr_region.name = new_name
+            with undo.group("Set Name"):
+                self.curr_region.set_name(new_name)
+                self.layer.set_gui_selected_region(self.curr_region)
```

Comment:
Is this necessary?

Reply:
It is a neat way to show region where the attribute was changed when undoing/redoing. note: `set_gui_selected_region` is undoable method.

Reply:
I added comments to each occurence of this trick.